pinot-core/src/test/java/org/apache/pinot/core/query/aggregation/function/DistinctAggregationFunctionTest.java

```diff
@@ -0,0 +1,281 @@
+            "null",
+            "1",
+            "2"
+        ).andOnSecondInstance("myField",
+            "2",
+            "3",
+            "2"
+        ).whenQuery("select 'literal', sum(distinct myField) from testTable group by 'literal'")
+        .thenResultIs("STRING | DOUBLE", "literal | " + addToDefaultNullValue(scenario.getDataType(), 6));
+  }
+
+  @Test(dataProvider = "scenarios")
+  void distinctSumAggregationGroupBySVWithNullHandlingEnabled(DataTypeScenario scenario) {
+    scenario.getDeclaringTable(true)
+        .onFirstInstance("myField",
+            "null",
+            "1",
+            "2"
+        ).andOnSecondInstance("myField",
+            "2",
+            "3",
+            "2"
+        ).whenQuery("select 'literal', sum(distinct myField) from testTable group by 'literal'")
+        .thenResultIs("STRING | DOUBLE", "literal | 6");
+  }
+
+  @Test(dataProvider = "scenarios")
+  void distinctAvgAggregationWithNullHandlingDisabled(DataTypeScenario scenario) {
+    scenario.getDeclaringTable(false, FieldSpec.FieldType.METRIC)
+        .onFirstInstance("myField",
+            "null",
+            "1",
+            "2"
+        ).andOnSecondInstance("myField",
+            "2",
+            "null",
+            "null"
+        ).whenQuery("select avg(distinct myField) from testTable")
+        .thenResultIs("DOUBLE", "1.0");
+  }
+
+  @Test(dataProvider = "scenarios")
+  void distinctAvgAggregationWithNullHandlingEnabled(DataTypeScenario scenario) {
+    scenario.getDeclaringTable(true, FieldSpec.FieldType.METRIC)
+        .onFirstInstance("myField",
+            "null",
+            "1",
+            "2"
+        ).andOnSecondInstance("myField",
+            "2",
+            "null",
+            "null"
+        ).whenQuery("select avg(distinct myField) from testTable")
+        .thenResultIs("DOUBLE", "1.5");
+  }
+
+  @Test(dataProvider = "scenarios")
+  void distinctAvgAggregationGroupBySVWithNullHandlingDisabled(DataTypeScenario scenario) {
+    scenario.getDeclaringTable(false, FieldSpec.FieldType.METRIC)
+        .onFirstInstance("myField",
+            "null",
+            "1",
+            "2"
+        ).andOnSecondInstance("myField",
+            "2",
+            "3",
+            "2"
+        ).whenQuery("select 'literal', avg(distinct myField) from testTable group by 'literal'")
+        .thenResultIs("STRING | DOUBLE", "literal | 1.5");
+  }
+
+  @Test(dataProvider = "scenarios")
+  void distinctAvgAggregationGroupBySVWithNullHandlingEnabled(DataTypeScenario scenario) {
+    scenario.getDeclaringTable(true, FieldSpec.FieldType.METRIC)
+        .onFirstInstance("myField",
+            "null",
+            "1",
+            "2"
+        ).andOnSecondInstance("myField",
+            "2",
+            "3",
+            "2"
+        ).whenQuery("select 'literal', avg(distinct myField) from testTable group by 'literal'")
+        .thenResultIs("STRING | DOUBLE", "literal | 2.0");
+  }
```

Comment:
idea (not for this PR): It is very verbose to have to repeat almost everything in order to test with and without null handling. We should think about adding something that let us write a single test like:

```java
    scenario.getDeclaringTable(FieldSpec.FieldType.METRIC) // this should ingest nulls
        .onFirstInstance("myField",
            "null",
            "1",
            "2"
        ).andOnSecondInstance("myField",
            "2",
            "3",
            "2"
        ).whenQuery("select 'literal', avg(distinct myField) from testTable group by 'literal'")
        .thenResultIs("STRING | DOUBLE", "literal | 2.0")
        .andWithoutNullHandlingIs("STRING | DOUBLE", "literal | 1.5");
```